ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -611,6 +611,19 @@ public func single<T, E>(producer: SignalProducer<T, E>) -> Result<T, E>? {
 	return result
 }
 
+/// Starts the producer, then blocks, waiting for all values.
+public func allValues<T, E>(producer: SignalProducer<T, E>) -> Result<[T], E> {
+	let resultOrNil = producer
+		|> reduce([T]()) { (var previous, value) in
```

Comment:
Maybe I should extract this into `collect` like in `RACSignal`? Then make `allValues` simply add `single`.